src/Extensions/SiteTreeSubsites.php

```diff
@@ -524,23 +515,26 @@ public function augmentSyncLinkTracking()
      */
     public function augmentValidURLSegment()
     {
-        // If this page is being filtered in the current subsite, then no custom validation query is required.
-        $subsite = Subsite::$force_subsite ?: SubsiteState::singleton()->getSubsiteId();
-        if (empty($this->owner->SubsiteID) || $subsite == $this->owner->SubsiteID) {
-            return null;
-        }
-
-        // Backup forced subsite
-        $prevForceSubsite = Subsite::$force_subsite;
-        Subsite::$force_subsite = $this->owner->SubsiteID;
-
-        // Repeat validation in the correct subsite
-        $isValid = $this->owner->validURLSegment();
-
-        // Restore
-        Subsite::$force_subsite = $prevForceSubsite;
-
-        return (bool)$isValid;
+        return SubsiteState::singleton()->withState(function (SubsiteState $newState) {
+            
+            // If this page is being filtered in the current subsite, then no custom validation query is required.
+            $subsite = $newState->getSubsiteId() ?: SubsiteState::singleton()->getSubsiteId();
+            if (empty($this->owner->SubsiteID) || $subsite == $this->owner->SubsiteID) {
+                return null;
+            }
+            
+            // Backup forced subsite
+            $prevForceSubsite = $newState->getSubsiteId();
```

Comment:
This seems wrong, we shouldn't need to repeat the previous logic backing up force_subsite inside the new withState() method, the point of the with*() pattern of methods is they reset themselves

Reply:
DONE